src/test/java/org/cactoos/time/DateAsTextTest.java

```diff
@@ -0,0 +1,168 @@
+        throws Exception {
+        final ZonedDateTime date = ZonedDateTime.of(
+            2017, 12, 13, 14, 15, 16, 17, ZoneId.of("Europe/Berlin")
+        );
+        MatcherAssert.assertThat(
+            "Can't format a ZonedDateTime with format.",
+            new DateAsText(date, "yyyy-MM-dd HH:mm:ss").asString(),
+            Matchers.is("2017-12-13 14:15:16")
+        );
+    }
+
+    @Test
+    public final void testZonedDateTimeFormattedWithFormatStringWithLocale()
+        throws Exception {
+        final ZonedDateTime date = ZonedDateTime.of(
+            2017, 12, 13, 14, 15, 16, 17, ZoneId.of("Europe/Berlin")
+        );
+        MatcherAssert.assertThat(
+            "Can't format a ZonedDateTime with format using locale.",
+            new DateAsText(date, "yyyy MMM dd. HH.mm.ss", Locale.FRENCH)
+                .asString(),
+            Matchers.is("2017 déc. 13. 14.15.16")
+        );
+    }
+
+    @Test
+    public final void testDateFormattedUsingIsoFormatter() throws Exception {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(2017, Calendar.DECEMBER, 13, 14, 15, 16);
+        calendar.set(Calendar.MILLISECOND, 17);
+        MatcherAssert.assertThat(
+            "Can't format a java.util.Date with ISO format.",
+            new DateAsText(calendar.getTime()).asString(),
+            Matchers.is("2017-12-13T14:15:16.017")
+        );
+    }
+
+    @Test
+    public final void testDateFormattedUsingCustomFormat() throws Exception {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(2017, Calendar.DECEMBER, 13, 14, 15, 16);
+        MatcherAssert.assertThat(
+            "Can't format a java.util.Date with custom format.",
+            new DateAsText(
+                calendar.getTime(), "yyyy MM dd hh:mm:ss"
+            ).asString(),
+            Matchers.is("2017 12 13 02:15:16")
+        );
+    }
+
+    @Test
+    public final void testDateFormattedUsingCustomFormatDifferentLocale()
+        throws Exception {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(2017, Calendar.DECEMBER, 13, 14, 15, 16);
+        MatcherAssert.assertThat(
+            "Can't format a java.util.Date with custom format.",
+            new DateAsText(
+                calendar.getTime(), "yyyy MMM dd hh:mm:ss", Locale.ITALIAN
+            ).asString(),
+            Matchers.is("2017 dic 13 02:15:16")
+        );
+    }
+
+    @Test
+    public final void testMillisFormattedUsingIsoFormatter()
+        throws Exception {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(2017, Calendar.DECEMBER, 13, 14, 15, 16);
+        calendar.set(Calendar.MILLISECOND, 17);
+        MatcherAssert.assertThat(
+            "Can't format a java.util.Date with ISO format.",
+            new DateAsText(calendar.getTime().getTime()).asString(),
+            Matchers.is("2017-12-13T14:15:16.017")
```

Comment:
@svendiedrichsen same here, timezone is missing

Reply:
The long milliseconds is the same story as with the Date. As the Date is simply a construct around the milliseconds instant.